internal/parser/parser.go

```diff
@@ -132,3 +138,21 @@ func structMethods(namedType *types.Named) ([]Method, error) {
 
 	return methods, nil
 }
+
+func parseType(t types.Type) Type {
+	// Remove every qualifier before the type name
+	// transforming "github.com/google/uuid.UUID" into "UUID"
+	noPackageQualifier := func(p *types.Package) string { return "" }
+
+	// Adds the package qualifier before the type name
+	// transforming "github.com/google/uuid.UUID" into "uuid.UUID"
+	relativeToExternalPackageQualifier := func(pkg *types.Package) string {
+		return pkg.Name()
+	}
+
+	return Type{
+		Name:         types.TypeString(t, relativeToExternalPackageQualifier), // (e.g. "uuid.UUID")
+		InternalName: types.TypeString(t, noPackageQualifier),                 // (e.g. "UUID")
```

Comment:
InternalName is the variable to use when you generate Go code **inside** the scanned package.
Name is the variable to use when you generate Go code **outside** the scanned package